pkg/backend/display/json.go

```diff
@@ -221,6 +230,8 @@ type previewStep struct {
 	DiffReasons []resource.PropertyKey `json:"diffReasons,omitempty"`
 	// ReplaceReasons is a list of keys that are causing replacement (for replacement steps only).
 	ReplaceReasons []resource.PropertyKey `json:"replaceReasons,omitempty"`
+	// RichDiff is a structured diff that indicates precise per-property differences.
+	RichDiff map[string]string `json:"richDiff"`
```

Comment:
if it's a per-property diff, i would prefer PropertyDiffs.  That would help indicate what the key is.

Reply:
I've decided to move forward with `DetailedDiff`.